internal/worker/workermanager.go

```diff
@@ -38,8 +38,9 @@ func (wm *WorkerManager) RegisterWorker(worker Worker) error {
 
 	wm.connectedClients.Store(worker.ID(), worker)
 	respChan := worker.(*BaseWorker).respChan
-	if respChan != nil {
-		wm.shardManager.RegisterWorker(worker.ID(), respChan) // TODO: Change respChan type to ShardResponse
+	preProcessingChan := worker.(*BaseWorker).preProcessingChan
+	if respChan != nil && preProcessingChan != nil {
+		wm.shardManager.RegisterWorker(worker.ID(), respChan, preProcessingChan) // TODO: Change respChan type to ShardResponse
 	}
```

Comment:
is the nil case only possible while we are in the migration process? If yes, then let's add a todo here to remove the conditional worker registration and instead return errors if either of these channels is not passed.

Reply:
This will stay there as long as we need preprocessing step. Still I have updated conditions, please suggest the changes if you feel.